File: core/exceptions.rb
module Exceptions
  # DONT ROLLBACK StandardError
  # DO ROLLBACK all NotMatching
NotMatching = Class.new StandardError
NotPassing = Class.new StandardError
NoResult = Class.new NotMatching
EndOfDocument = Class.new StandardError
EndOfLine=  Class.new NotMatching
MaxRecursionReached= Class.new StandardError
EndOfBlock=  Class.new NotMatching
GivingUp= Class.new StandardError
ShouldNotMatchKeyword=  Class.new NotMatching
KeywordNotExpected=  Class.new NotMatching
UndefinedRubyMethod=  Class.new NotMatching
WrongType = Class.new StandardError
end

File: angelos/core/exceptions.py
# encoding: utf-8
# import minitest
  # DONT ROLLBACK StandardError
  # DO ROLLBACK all NotMatching
global NotMatching
# class StandardError(Exception):
#     pass

# class Error(Exception):
#     pass
class InternalError(StandardError):
    pass

class NoMethodError(StandardError):
    pass

class InternalError(StandardError):
    pass

class NotMatching(StandardError):
    pass

class UnknownCommandError(StandardError):
    pass

class SecurityError(StandardError):
    pass

# NotPassing = Class.new StandardError
class NotPassing(StandardError):
    pass

class NoResult(NotMatching):
    pass

class EndOfDocument(StandardError):
    pass

class EndOfLine(NotMatching):
    pass

class MaxRecursionReached(StandardError):
    pass

class EndOfBlock(NotMatching):
    pass

class GivingUp(StandardError):
    pass

class ShouldNotMatchKeyword(NotMatching):
    pass

class KeywordNotExpected(NotMatching):
    pass

class UndefinedRubyMethod(NotMatching):
    pass

class WrongType(StandardError):
    pass

class ImmutableVaribale(StandardError):
    pass

class SystemStackError(StandardError):
    pass



